src/SNLS_qr_solve.h

```diff
@@ -0,0 +1,204 @@
+#pragma once
+
+#include "SNLS_base.h"
+#include "SNLS_linalg.h"
+
+#include <stdlib.h>
+#include <iostream>
+#ifdef __cuda_host_only__
+#include <string>
+#include <sstream>
+#include <iomanip>
+#endif
+
+namespace snls {
+
+// Need to add an actual QR solver in here at some point. It currently just contains just QR factorization functions
+
+// Algorithm taken from Matrix Computation 4th ed. by GH Golub and CF Van Loan
+// pg. 236 alg. 5.1.1
+__snls_hdev__
+inline
+void houseHolderVec(double& beta,
```

Comment:
Given that Householder is all one word, here and elsewhere you probably do not want to capitalize the "Holder" part of householder when doing camelCase.